### ISSUE TITLE
Output probelm of step_outliers_outForest

### ISSUE THREAD

**Issue by xiaochi-liu** [brunocarlin/tidy.outliers#21]
Hello, it seems that there's a problem with the output of `step_outliers_outForest`.

When the code is run by `reprex()`, everything is fine:

``` r
library(recipes)
#> Loading required package: dplyr
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union
#> 
#> Attaching package: 'recipes'
#> The following object is masked from 'package:stats':
#> 
#>     step
library(tidy.outliers)

rec <-
  recipe(mpg ~ ., data = mtcars) %>%
  step_outliers_outForest(all_numeric_predictors()) %>%
  prep(mtcars)
#> Due to small sample size, reduced 'min.node.size' to 11

bake(rec, new_data = NULL) %>% 
  select(.outliers_outForest)
#> # A tibble: 32 × 1
#>    .outliers_outForest$score
#>                        <dbl>
#>  1                         0
#>  2                         0
#>  3                         0
#>  4                         0
#>  5                         0
#>  6                         0
#>  7                         0
#>  8                         0
#>  9                         1
#> 10                         0
#> # … with 22 more rows
```

However, when the same code is run in `.rmd` file, the type of `.outliers_outForest` column is `tibble`:

![image](https://user-images.githubusercontent.com/48895627/208225339-967f0258-75d6-44b2-ac0c-8b35be75d1a4.png)

**Comment by brunocarlin** [comment]
Hey I am going on vacation, so no work will be done by me for the next 15 days, but feel free to ask simple questions

**Comment by brunocarlin** [comment]
The code will always return a tibble, even your reprex is a tible as you can see .outliers_outForest$score is just a nicer printing, but if you check types it will be a tibble.

**Comment by xiaochi-liu** [comment]
Many thanks for your response @brunocarlin!

It seems that if a tibble is returned, the next `step_outliers_remove` is going to collapse:

``` r
library(recipes)
#> Loading required package: dplyr
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union
#> 
#> Attaching package: 'recipes'
#> The following object is masked from 'package:stats':
#> 
#>     step
library(tidy.outliers)

rec <-
  recipe(mpg ~ ., data = mtcars) %>%
  step_outliers_outForest(all_numeric_predictors()) %>%
  step_outliers_remove(contains(".outliers")) %>% 
  prep(mtcars)
#> Due to small sample size, reduced 'min.node.size' to 11
#> Error in `check_type()`:
#> ! All columns selected for the step should be numeric.

#> Backtrace:
#>     ▆
#>  1. ├─... %>% prep(mtcars)
#>  2. ├─recipes::prep(., mtcars)
#>  3. └─recipes:::prep.recipe(., mtcars)
#>  4.   ├─recipes::prep(x$steps[[i]], training = training, info = x$term_info)
#>  5.   └─tidy.outliers:::prep.step_outliers_remove(...)
#>  6.     └─recipes::check_type(training[, col_names])
#>  7.       └─rlang::abort(...)
```

<sup>Created on 2023-01-12 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by brunocarlin** [comment]
Well this is bad


```
Warning message:
Returning more (or less) than 1 row per `summarise()` group was deprecated in dplyr 1.1.0.
ℹ Please use `reframe()` instead.
ℹ When switching from `summarise()` to `reframe()`, remember that `reframe()` always returns an ungrouped data frame and
  adjust accordingly.
ℹ The deprecated feature was likely used in the tidy.outliers package.
  Please report the issue at <https://github.com/brunocarlin/tidy.outliers/issues>.
This warning is displayed once every 8 hours.
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was generated. 
```

**Comment by brunocarlin** [comment]
The good news is that this seems related to only outForest, the bad news is that I have no idea what the problem is

**Comment by brunocarlin** [comment]
Ok so running on an old renv version it seems this was introduced with some package making an update, I will bet dplyr]

``` r
library(tidymodels)
#> Registered S3 method overwritten by 'tune':
#>   method                   from   
#>   required_pkgs.model_spec parsnip
library(tidy.outliers)

rec <-
  recipe(mpg ~ ., data = mtcars) %>%
  step_outliers_outForest(all_numeric_predictors()) %>%
  prep(mtcars)
#> Due to small sample size, reduced 'min.node.size' to 11

bake(rec, new_data = NULL) %>% 
  select(.outliers_outForest)
#> # A tibble: 32 × 1
#>    .outliers_outForest$score
#>                        <dbl>
#>  1                         0
#>  2                         0
#>  3                         0
#>  4                         0
#>  5                         0
#>  6                         0
#>  7                         0
#>  8                         0
#>  9                         1
#> 10                         0
#> # … with 22 more rows
```

<sup>Created on 2023-02-14 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

<details style="margin-bottom:10px;">
<summary>
Session info
</summary>

``` r
sessioninfo::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value
#>  version  R version 4.2.2 (2022-10-31 ucrt)
#>  os       Windows 10 x64 (build 22621)
#>  system   x86_64, mingw32
#>  ui       RTerm
#>  language (EN)
#>  collate  Portuguese_Brazil.utf8
#>  ctype    Portuguese_Brazil.utf8
#>  tz       America/Sao_Paulo
#>  date     2023-02-14
#>  pandoc   2.19.2 @ C:/Program Files/RStudio/resources/app/bin/quarto/bin/tools/ (via rmarkdown)
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  ! package       * version    date (UTC) lib source
#>  P backports       1.4.1      2021-12-13 [?] CRAN (R 4.2.0)
#>  P bitops          1.0-7      2021-04-24 [?] CRAN (R 4.2.0)
#>  P broom         * 0.7.12     2022-01-28 [?] CRAN (R 4.2.2)
#>  P class           7.3-20     2022-01-16 [?] CRAN (R 4.2.2)
#>    cli             3.6.0      2023-01-09 [1] CRAN (R 4.2.2)
#>  P codetools       0.2-18     2020-11-04 [?] CRAN (R 4.2.0)
#>  P colorspace      2.0-3      2022-02-21 [?] CRAN (R 4.2.2)
#>  P crayon          1.5.0      2022-02-14 [?] CRAN (R 4.2.2)
#>  P dials         * 0.1.0      2022-01-31 [?] CRAN (R 4.2.2)
#>  P DiceDesign      1.9        2021-02-13 [?] CRAN (R 4.2.2)
#>  P digest          0.6.29     2021-12-01 [?] CRAN (R 4.2.1)
#>  P dplyr         * 1.0.8      2022-02-08 [?] CRAN (R 4.2.2)
#>  P ellipsis        0.3.2      2021-04-29 [?] CRAN (R 4.2.2)
#>  P evaluate        0.15       2022-02-18 [?] CRAN (R 4.2.1)
#>  P fansi           1.0.2      2022-01-14 [?] CRAN (R 4.2.2)
#>  P fastmap         1.1.0      2021-01-25 [?] CRAN (R 4.2.2)
#>  P FNN             1.1.3      2019-02-15 [?] CRAN (R 4.2.0)
#>  P foreach         1.5.2      2022-02-02 [?] CRAN (R 4.2.2)
#>  P fs              1.5.2      2021-12-08 [?] CRAN (R 4.2.2)
#>  P furrr           0.2.3      2021-06-25 [?] CRAN (R 4.2.2)
#>  P future          1.24.0     2022-02-19 [?] CRAN (R 4.2.2)
#>  P future.apply    1.8.1      2021-08-10 [?] CRAN (R 4.2.2)
#>  P generics        0.1.2      2022-01-31 [?] CRAN (R 4.2.1)
#>  P ggplot2       * 3.3.5      2021-06-25 [?] CRAN (R 4.2.2)
#>  P globals         0.14.0     2020-11-22 [?] CRAN (R 4.2.2)
#>  P glue            1.6.2      2022-02-24 [?] CRAN (R 4.2.2)
#>  P gower           1.0.0      2022-02-03 [?] CRAN (R 4.2.0)
#>  P GPfit           1.0-8      2019-02-08 [?] CRAN (R 4.2.2)
#>  P gtable          0.3.0      2019-03-25 [?] CRAN (R 4.2.1)
#>  P h2o             3.36.0.3   2022-02-17 [?] CRAN (R 4.2.2)
#>  P hardhat         0.2.0      2022-01-24 [?] CRAN (R 4.2.0)
#>  P highr           0.9        2021-04-16 [?] CRAN (R 4.2.2)
#>    htmltools       0.5.4      2022-12-07 [1] CRAN (R 4.2.2)
#>  P infer         * 1.0.0      2021-08-13 [?] CRAN (R 4.2.0)
#>  P ipred           0.9-12     2021-09-15 [?] CRAN (R 4.2.0)
#>  P iterators       1.0.14     2022-02-05 [?] CRAN (R 4.2.2)
#>  P jsonlite        1.8.0      2022-02-22 [?] CRAN (R 4.2.1)
#>  P knitr           1.37       2021-12-16 [?] CRAN (R 4.2.2)
#>  P lattice         0.20-45    2021-09-22 [3] CRAN (R 4.2.2)
#>  P lava            1.6.10     2021-09-02 [?] CRAN (R 4.2.1)
#>  P lhs             1.1.4      2022-02-20 [?] CRAN (R 4.2.2)
#>  P lifecycle       1.0.1      2021-09-24 [?] CRAN (R 4.2.1)
#>  P listenv         0.8.0      2019-12-05 [?] CRAN (R 4.2.2)
#>  P lookout         0.1.1      2023-02-15 [?] Github (sevvandi/lookout@21c41bc)
#>  P lubridate       1.8.0      2021-10-07 [?] CRAN (R 4.2.2)
#>  P magrittr        2.0.2      2022-01-26 [?] CRAN (R 4.2.2)
#>  P MASS            7.3-55     2022-01-13 [?] CRAN (R 4.2.2)
#>  P Matrix          1.4-0      2021-12-08 [?] CRAN (R 4.2.2)
#>  P missRanger      2.1.3      2021-03-30 [?] CRAN (R 4.2.2)
#>  P modeldata     * 0.1.1      2021-07-14 [?] CRAN (R 4.2.1)
#>  P munsell         0.5.0      2018-06-12 [?] CRAN (R 4.2.2)
#>  P nnet            7.3-17     2022-01-16 [?] CRAN (R 4.2.1)
#>  P outForest       0.1.2      2022-01-31 [?] CRAN (R 4.2.2)
#>  P parallelly      1.30.0     2021-12-17 [?] CRAN (R 4.2.2)
#>  P parsnip       * 0.1.7      2021-07-21 [?] CRAN (R 4.2.2)
#>  P pillar          1.7.0      2022-02-01 [?] CRAN (R 4.2.1)
#>  P pkgconfig       2.0.3      2019-09-22 [?] CRAN (R 4.2.2)
#>  P plyr            1.8.6      2020-03-03 [?] CRAN (R 4.2.2)
#>  P pROC            1.18.0     2021-09-03 [?] CRAN (R 4.2.2)
#>  P prodlim         2019.11.13 2019-11-17 [?] CRAN (R 4.2.2)
#>  P purrr         * 0.3.4      2020-04-17 [?] CRAN (R 4.2.1)
#>  P R6              2.5.1      2021-08-19 [?] CRAN (R 4.2.2)
#>  P ranger          0.13.1     2021-07-14 [?] CRAN (R 4.2.0)
#>  P Rcpp            1.0.8      2022-01-13 [?] CRAN (R 4.2.2)
#>  P RCurl           1.98-1.6   2022-02-08 [?] CRAN (R 4.2.0)
#>  P recipes       * 0.2.0      2022-02-18 [?] CRAN (R 4.2.1)
#>    reprex          2.0.2      2022-08-17 [1] CRAN (R 4.2.2)
#>    rlang           1.0.6      2022-09-24 [1] CRAN (R 4.2.2)
#>  P rmarkdown       2.12       2022-03-02 [?] CRAN (R 4.2.2)
#>  P rpart           4.1.16     2022-01-24 [?] CRAN (R 4.2.1)
#>  P rsample       * 0.1.1      2021-11-08 [?] CRAN (R 4.2.0)
#>  P rstudioapi      0.13       2020-11-12 [?] CRAN (R 4.2.1)
#>  P scales        * 1.1.1      2020-05-11 [?] CRAN (R 4.2.2)
#>    sessioninfo     1.2.2      2021-12-06 [1] CRAN (R 4.2.2)
#>  P stringi         1.7.6      2021-11-29 [?] CRAN (R 4.2.0)
#>  P stringr         1.4.0      2019-02-10 [?] CRAN (R 4.2.1)
#>  P survival        3.3-1      2022-03-03 [?] CRAN (R 4.2.1)
#>  P tibble        * 3.1.6      2021-11-07 [?] CRAN (R 4.2.2)
#>    tidy.outliers * 0.2.0      2023-02-15 [1] local
#>  P tidymodels    * 0.1.4      2021-10-01 [?] CRAN (R 4.2.2)
#>  P tidyr         * 1.2.0      2022-02-01 [?] CRAN (R 4.2.1)
#>  P tidyselect      1.1.2      2022-02-21 [?] CRAN (R 4.2.1)
#>  P timeDate        3043.102   2018-02-21 [?] CRAN (R 4.2.0)
#>  P tune          * 0.1.6      2021-07-21 [?] CRAN (R 4.2.2)
#>  P utf8            1.2.2      2021-07-24 [?] CRAN (R 4.2.2)
#>  P vctrs           0.3.8      2021-04-29 [?] CRAN (R 4.2.2)
#>  P withr           2.5.0      2022-03-03 [?] CRAN (R 4.2.2)
#>  P workflows     * 0.2.4      2021-10-12 [?] CRAN (R 4.2.2)
#>  P workflowsets  * 0.1.0      2021-07-22 [?] CRAN (R 4.2.2)
#>  P xfun            0.30       2022-03-02 [?] CRAN (R 4.2.2)
#>  P yaml            2.3.5      2022-02-21 [?] CRAN (R 4.2.1)
#>  P yardstick     * 0.0.9      2021-11-22 [?] CRAN (R 4.2.0)
#> 
#>  [1] C:/Users/bruno/AppData/Local/R/cache/R/renv/library/tidy.outliers-e7ed4b7d/R-4.2/x86_64-w64-mingw32
#>  [2] C:/Users/bruno/AppData/Local/Temp/RtmpCgAaCX/renv-system-library
#>  [3] C:/Program Files/R/R-4.2.2/library
#> 
#>  P ── Loaded and on-disk path mismatch.
#> 
#> ──────────────────────────────────────────────────────────────────────────────
```

</details>

**Comment by brunocarlin** [comment]
Ok this was right, as soon as I update all packages

``` r
library(tidymodels)
library(tidy.outliers)

rec <-
  recipe(mpg ~ ., data = mtcars) %>%
  step_outliers_outForest(all_numeric_predictors()) %>%
  prep(mtcars)
#> Due to small sample size, reduced 'min.node.size' to 11
#> Warning: Returning more (or less) than 1 row per `summarise()` group was deprecated in
#> dplyr 1.1.0.
#> ℹ Please use `reframe()` instead.
#> ℹ When switching from `summarise()` to `reframe()`, remember that `reframe()`
#>   always returns an ungrouped data frame and adjust accordingly.
#> ℹ The deprecated feature was likely used in the tidy.outliers package.
#>   Please report the issue at
#>   <https://github.com/brunocarlin/tidy.outliers/issues>.

bake(rec, new_data = NULL) %>% 
  select(.outliers_outForest)
#> # A tibble: 32 × 1
#>    .outliers_outForest$score
#>                        <dbl>
#>  1                         0
#>  2                         0
#>  3                         0
#>  4                         0
#>  5                         0
#>  6                         0
#>  7                         0
#>  8                         0
#>  9                         1
#> 10                         0
#> # … with 22 more rows
```

<sup>Created on 2023-02-14 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

<details style="margin-bottom:10px;">
<summary>
Session info
</summary>

``` r
sessioninfo::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value
#>  version  R version 4.2.2 (2022-10-31 ucrt)
#>  os       Windows 10 x64 (build 22621)
#>  system   x86_64, mingw32
#>  ui       RTerm
#>  language (EN)
#>  collate  Portuguese_Brazil.utf8
#>  ctype    Portuguese_Brazil.utf8
#>  tz       America/Sao_Paulo
#>  date     2023-02-14
#>  pandoc   2.19.2 @ C:/Program Files/RStudio/resources/app/bin/quarto/bin/tools/ (via rmarkdown)
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  ! package       * version    date (UTC) lib source
#>  P backports       1.4.1      2021-12-13 [?] CRAN (R 4.2.0)
#>  P bitops          1.0-7      2021-04-24 [?] CRAN (R 4.2.0)
#>    broom         * 1.0.3      2023-01-25 [1] CRAN (R 4.2.2)
#>    class           7.3-21     2023-01-23 [1] CRAN (R 4.2.2)
#>    cli             3.6.0      2023-01-09 [1] CRAN (R 4.2.2)
#>    codetools       0.2-19     2023-02-01 [1] CRAN (R 4.2.2)
#>    colorspace      2.1-0      2023-01-23 [1] CRAN (R 4.2.2)
#>    dials         * 1.1.0      2022-11-04 [1] CRAN (R 4.2.2)
#>  P DiceDesign      1.9        2021-02-13 [?] CRAN (R 4.2.2)
#>    digest          0.6.31     2022-12-11 [1] CRAN (R 4.2.2)
#>    dplyr         * 1.1.0      2023-01-29 [1] CRAN (R 4.2.2)
#>  P ellipsis        0.3.2      2021-04-29 [?] CRAN (R 4.2.2)
#>    evaluate        0.20       2023-01-17 [1] CRAN (R 4.2.2)
#>    fansi           1.0.4      2023-01-22 [1] CRAN (R 4.2.2)
#>  P fastmap         1.1.0      2021-01-25 [?] CRAN (R 4.2.2)
#>    FNN             1.1.3.1    2022-05-23 [1] CRAN (R 4.2.2)
#>  P foreach         1.5.2      2022-02-02 [?] CRAN (R 4.2.2)
#>    fs              1.6.1      2023-02-06 [1] CRAN (R 4.2.2)
#>    furrr           0.3.1      2022-08-15 [1] CRAN (R 4.2.2)
#>    future          1.31.0     2023-02-01 [1] CRAN (R 4.2.2)
#>    future.apply    1.10.0     2022-11-05 [1] CRAN (R 4.2.2)
#>    generics        0.1.3      2022-07-05 [1] CRAN (R 4.2.2)
#>    ggplot2       * 3.4.1      2023-02-10 [1] CRAN (R 4.2.2)
#>    globals         0.16.2     2022-11-21 [1] CRAN (R 4.2.2)
#>  P glue            1.6.2      2022-02-24 [?] CRAN (R 4.2.2)
#>    gower           1.0.1      2022-12-22 [1] CRAN (R 4.2.2)
#>  P GPfit           1.0-8      2019-02-08 [?] CRAN (R 4.2.2)
#>    gtable          0.3.1      2022-09-01 [1] CRAN (R 4.2.2)
#>    h2o             3.38.0.1   2022-09-23 [1] CRAN (R 4.2.2)
#>    hardhat         1.2.0      2022-06-30 [1] CRAN (R 4.2.2)
#>    htmltools       0.5.4      2022-12-07 [1] CRAN (R 4.2.2)
#>    infer         * 1.0.4      2022-12-02 [1] CRAN (R 4.2.2)
#>    ipred           0.9-13     2022-06-02 [1] CRAN (R 4.2.2)
#>  P iterators       1.0.14     2022-02-05 [?] CRAN (R 4.2.2)
#>    jsonlite        1.8.4      2022-12-06 [1] CRAN (R 4.2.2)
#>    knitr           1.42       2023-01-25 [1] CRAN (R 4.2.2)
#>  P lattice         0.20-45    2021-09-22 [3] CRAN (R 4.2.2)
#>    lava            1.7.1      2023-01-06 [1] CRAN (R 4.2.2)
#>    lhs             1.1.6      2022-12-17 [1] CRAN (R 4.2.2)
#>    lifecycle       1.0.3      2022-10-07 [1] CRAN (R 4.2.2)
#>    listenv         0.9.0      2022-12-16 [1] CRAN (R 4.2.2)
#>    lookout         0.1.4      2022-10-14 [1] CRAN (R 4.2.2)
#>    lubridate       1.9.2      2023-02-10 [1] CRAN (R 4.2.2)
#>    magrittr        2.0.3      2022-03-30 [1] CRAN (R 4.2.2)
#>    MASS            7.3-58.2   2023-01-23 [1] CRAN (R 4.2.2)
#>    Matrix          1.5-3      2022-11-11 [1] CRAN (R 4.2.2)
#>  P missRanger      2.1.3      2021-03-30 [?] CRAN (R 4.2.2)
#>    modeldata     * 1.1.0      2023-01-25 [1] CRAN (R 4.2.2)
#>  P munsell         0.5.0      2018-06-12 [?] CRAN (R 4.2.2)
#>    nnet            7.3-18     2022-09-28 [1] CRAN (R 4.2.2)
#>  P outForest       0.1.2      2022-01-31 [?] CRAN (R 4.2.2)
#>    parallelly      1.34.0     2023-01-13 [1] CRAN (R 4.2.2)
#>    parsnip       * 1.0.3      2022-11-11 [1] CRAN (R 4.2.2)
#>    pillar          1.8.1      2022-08-19 [1] CRAN (R 4.2.2)
#>  P pkgconfig       2.0.3      2019-09-22 [?] CRAN (R 4.2.2)
#>  P prodlim         2019.11.13 2019-11-17 [?] CRAN (R 4.2.2)
#>    purrr         * 1.0.1      2023-01-10 [1] CRAN (R 4.2.2)
#>  P R6              2.5.1      2021-08-19 [?] CRAN (R 4.2.2)
#>    ranger          0.14.1     2022-06-18 [1] CRAN (R 4.2.2)
#>    Rcpp            1.0.10     2023-01-22 [1] CRAN (R 4.2.2)
#>    RCurl           1.98-1.10  2023-01-27 [1] CRAN (R 4.2.2)
#>    recipes       * 1.0.4      2023-01-11 [1] CRAN (R 4.2.2)
#>    reprex          2.0.2      2022-08-17 [1] CRAN (R 4.2.2)
#>    rlang           1.0.6      2022-09-24 [1] CRAN (R 4.2.2)
#>    rmarkdown       2.20       2023-01-19 [1] CRAN (R 4.2.2)
#>    rpart           4.1.19     2022-10-21 [1] CRAN (R 4.2.2)
#>    rsample       * 1.1.1      2022-12-07 [1] CRAN (R 4.2.2)
#>    rstudioapi      0.14       2022-08-22 [1] CRAN (R 4.2.2)
#>    scales        * 1.2.1      2022-08-20 [1] CRAN (R 4.2.2)
#>    sessioninfo     1.2.2      2021-12-06 [1] CRAN (R 4.2.2)
#>    survival        3.5-3      2023-02-12 [1] CRAN (R 4.2.2)
#>    tibble        * 3.1.8      2022-07-22 [1] CRAN (R 4.2.2)
#>    tidy.outliers * 0.2.0      2023-02-15 [1] local
#>    tidymodels    * 1.0.0      2022-07-13 [1] CRAN (R 4.2.2)
#>    tidyr         * 1.3.0      2023-01-24 [1] CRAN (R 4.2.2)
#>    tidyselect      1.2.0      2022-10-10 [1] CRAN (R 4.2.2)
#>    timechange      0.2.0      2023-01-11 [1] CRAN (R 4.2.2)
#>    timeDate        4022.108   2023-01-07 [1] CRAN (R 4.2.2)
#>    tune          * 1.0.1      2022-10-09 [1] CRAN (R 4.2.2)
#>    utf8            1.2.3      2023-01-31 [1] CRAN (R 4.2.2)
#>    vctrs           0.5.2      2023-01-23 [1] CRAN (R 4.2.2)
#>  P withr           2.5.0      2022-03-03 [?] CRAN (R 4.2.2)
#>    workflows     * 1.1.2      2022-11-16 [1] CRAN (R 4.2.2)
#>    workflowsets  * 1.0.0      2022-07-12 [1] CRAN (R 4.2.2)
#>    xfun            0.37       2023-01-31 [1] CRAN (R 4.2.2)
#>    yaml            2.3.7      2023-01-23 [1] CRAN (R 4.2.2)
#>    yardstick     * 1.1.0      2022-09-07 [1] CRAN (R 4.2.2)
#> 
#>  [1] C:/Users/bruno/AppData/Local/R/cache/R/renv/library/tidy.outliers-e7ed4b7d/R-4.2/x86_64-w64-mingw32
#>  [2] C:/Users/bruno/AppData/Local/Temp/RtmpyOAe00/renv-system-library
#>  [3] C:/Program Files/R/R-4.2.2/library
#> 
#>  P ── Loaded and on-disk path mismatch.
#> 
#> ──────────────────────────────────────────────────────────────────────────────
```

</details>

**Comment by brunocarlin** [comment]
Turns out I even get a failure when running my tests

![image](https://user-images.githubusercontent.com/1176377/218893903-57c6d8c8-5c10-4f39-a523-f690072c71cb.png)

**Comment by brunocarlin** [comment]
Fixed with #24 

``` r
library(tidymodels)
library(tidy.outliers)

rec <-
  recipe(mpg ~ ., data = mtcars) %>%
  step_outliers_outForest(all_numeric_predictors()) %>%
  prep(mtcars)
#> Due to small sample size, reduced 'min.node.size' to 11

bake(rec, new_data = NULL) %>%
  select(contains('out'))
#> # A tibble: 32 × 1
#>    .outliers_outForest
#>                  <dbl>
#>  1                   0
#>  2                   0
#>  3                   0
#>  4                   0
#>  5                   0
#>  6                   0
#>  7                   0
#>  8                   0
#>  9                   1
#> 10                   0
#> # … with 22 more rows
```

<sup>Created on 2023-02-14 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

<details style="margin-bottom:10px;">
<summary>
Session info
</summary>

``` r
sessioninfo::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value
#>  version  R version 4.2.2 (2022-10-31 ucrt)
#>  os       Windows 10 x64 (build 22621)
#>  system   x86_64, mingw32
#>  ui       RTerm
#>  language (EN)
#>  collate  Portuguese_Brazil.utf8
#>  ctype    Portuguese_Brazil.utf8
#>  tz       America/Sao_Paulo
#>  date     2023-02-14
#>  pandoc   2.19.2 @ C:/Program Files/RStudio/resources/app/bin/quarto/bin/tools/ (via rmarkdown)
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package       * version    date (UTC) lib source
#>  backports       1.4.1      2021-12-13 [1] CRAN (R 4.2.0)
#>  bitops          1.0-7      2021-04-24 [1] CRAN (R 4.2.0)
#>  broom         * 1.0.3      2023-01-25 [1] CRAN (R 4.2.2)
#>  class           7.3-20     2022-01-16 [2] CRAN (R 4.2.2)
#>  cli             3.6.0      2023-01-09 [1] CRAN (R 4.2.2)
#>  codetools       0.2-18     2020-11-04 [2] CRAN (R 4.2.2)
#>  colorspace      2.1-0      2023-01-23 [1] CRAN (R 4.2.2)
#>  dials         * 1.1.0      2022-11-04 [1] CRAN (R 4.2.2)
#>  DiceDesign      1.9        2021-02-13 [1] CRAN (R 4.2.1)
#>  digest          0.6.31     2022-12-11 [1] CRAN (R 4.2.2)
#>  dplyr         * 1.1.0      2023-01-29 [1] CRAN (R 4.2.2)
#>  ellipsis        0.3.2      2021-04-29 [1] CRAN (R 4.2.1)
#>  evaluate        0.20       2023-01-17 [1] CRAN (R 4.2.2)
#>  fansi           1.0.4      2023-01-22 [1] CRAN (R 4.2.2)
#>  fastmap         1.1.0      2021-01-25 [1] CRAN (R 4.2.1)
#>  FNN             1.1.3.1    2022-05-23 [1] CRAN (R 4.2.1)
#>  foreach         1.5.2      2022-02-02 [1] CRAN (R 4.2.1)
#>  fs              1.6.1      2023-02-06 [1] CRAN (R 4.2.2)
#>  furrr           0.3.1      2022-08-15 [1] CRAN (R 4.2.1)
#>  future          1.31.0     2023-02-01 [1] CRAN (R 4.2.2)
#>  future.apply    1.10.0     2022-11-05 [1] CRAN (R 4.2.2)
#>  generics        0.1.3      2022-07-05 [1] CRAN (R 4.2.1)
#>  ggplot2       * 3.4.1      2023-02-10 [1] CRAN (R 4.2.2)
#>  globals         0.16.2     2022-11-21 [1] CRAN (R 4.2.2)
#>  glue            1.6.2      2022-02-24 [1] CRAN (R 4.2.1)
#>  gower           1.0.1      2022-12-22 [1] CRAN (R 4.2.2)
#>  GPfit           1.0-8      2019-02-08 [1] CRAN (R 4.2.1)
#>  gtable          0.3.1      2022-09-01 [1] CRAN (R 4.2.2)
#>  h2o             3.38.0.1   2022-09-23 [1] CRAN (R 4.2.2)
#>  hardhat         1.2.0      2022-06-30 [1] CRAN (R 4.2.1)
#>  htmltools       0.5.4      2022-12-07 [1] CRAN (R 4.2.2)
#>  infer         * 1.0.4      2022-12-02 [1] CRAN (R 4.2.2)
#>  ipred           0.9-13     2022-06-02 [1] CRAN (R 4.2.1)
#>  iterators       1.0.14     2022-02-05 [1] CRAN (R 4.2.1)
#>  jsonlite        1.8.4      2022-12-06 [1] CRAN (R 4.2.2)
#>  knitr           1.42       2023-01-25 [1] CRAN (R 4.2.2)
#>  lattice         0.20-45    2021-09-22 [2] CRAN (R 4.2.2)
#>  lava            1.7.1      2023-01-06 [1] CRAN (R 4.2.2)
#>  lhs             1.1.6      2022-12-17 [1] CRAN (R 4.2.2)
#>  lifecycle       1.0.3      2022-10-07 [1] CRAN (R 4.2.2)
#>  listenv         0.9.0      2022-12-16 [1] CRAN (R 4.2.2)
#>  lookout         0.1.4      2023-02-14 [1] Github (sevvandi/lookout@05cbd68)
#>  lubridate       1.9.2      2023-02-10 [1] CRAN (R 4.2.2)
#>  magrittr        2.0.3      2022-03-30 [1] CRAN (R 4.2.1)
#>  MASS            7.3-58.1   2022-08-03 [2] CRAN (R 4.2.2)
#>  Matrix          1.5-1      2022-09-13 [2] CRAN (R 4.2.2)
#>  missRanger      2.1.3      2021-03-30 [1] CRAN (R 4.2.2)
#>  modeldata     * 1.1.0      2023-01-25 [1] CRAN (R 4.2.2)
#>  munsell         0.5.0      2018-06-12 [1] CRAN (R 4.2.1)
#>  nnet            7.3-18     2022-09-28 [2] CRAN (R 4.2.2)
#>  outForest       0.1.2      2022-01-31 [1] CRAN (R 4.2.2)
#>  parallelly      1.34.0     2023-01-13 [1] CRAN (R 4.2.2)
#>  parsnip       * 1.0.3      2022-11-11 [1] CRAN (R 4.2.2)
#>  pillar          1.8.1      2022-08-19 [1] CRAN (R 4.2.1)
#>  pkgconfig       2.0.3      2019-09-22 [1] CRAN (R 4.2.1)
#>  prodlim         2019.11.13 2019-11-17 [1] CRAN (R 4.2.1)
#>  purrr         * 1.0.1      2023-01-10 [1] CRAN (R 4.2.2)
#>  R.cache         0.16.0     2022-07-21 [1] CRAN (R 4.2.1)
#>  R.methodsS3     1.8.2      2022-06-13 [1] CRAN (R 4.2.0)
#>  R.oo            1.25.0     2022-06-12 [1] CRAN (R 4.2.0)
#>  R.utils         2.12.2     2022-11-11 [1] CRAN (R 4.2.2)
#>  R6              2.5.1      2021-08-19 [1] CRAN (R 4.2.1)
#>  ranger          0.14.1     2022-06-18 [1] CRAN (R 4.2.1)
#>  Rcpp            1.0.10     2023-01-22 [1] CRAN (R 4.2.2)
#>  RCurl           1.98-1.10  2023-01-27 [1] CRAN (R 4.2.2)
#>  recipes       * 1.0.4      2023-01-11 [1] CRAN (R 4.2.2)
#>  reprex          2.0.2      2022-08-17 [1] CRAN (R 4.2.1)
#>  rlang           1.0.6      2022-09-24 [1] CRAN (R 4.2.2)
#>  rmarkdown       2.20       2023-01-19 [1] CRAN (R 4.2.2)
#>  rpart           4.1.19     2022-10-21 [2] CRAN (R 4.2.2)
#>  rsample       * 1.1.1      2022-12-07 [1] CRAN (R 4.2.2)
#>  rstudioapi      0.14       2022-08-22 [1] CRAN (R 4.2.1)
#>  scales        * 1.2.1      2022-08-20 [1] CRAN (R 4.2.1)
#>  sessioninfo     1.2.2      2021-12-06 [1] CRAN (R 4.2.1)
#>  styler          1.9.0      2023-01-15 [1] CRAN (R 4.2.2)
#>  survival        3.4-0      2022-08-09 [2] CRAN (R 4.2.2)
#>  tibble        * 3.1.8      2022-07-22 [1] CRAN (R 4.2.1)
#>  tidy.outliers * 0.3.0      2023-02-15 [1] Github (brunocarlin/tidy.outliers@7c2715a)
#>  tidymodels    * 1.0.0      2022-07-13 [1] CRAN (R 4.2.2)
#>  tidyr         * 1.3.0      2023-01-24 [1] CRAN (R 4.2.2)
#>  tidyselect      1.2.0      2022-10-10 [1] CRAN (R 4.2.2)
#>  timechange      0.2.0      2023-01-11 [1] CRAN (R 4.2.2)
#>  timeDate        4022.108   2023-01-07 [1] CRAN (R 4.2.2)
#>  tune          * 1.0.1      2022-10-09 [1] CRAN (R 4.2.2)
#>  utf8            1.2.3      2023-01-31 [1] CRAN (R 4.2.2)
#>  vctrs           0.5.2      2023-01-23 [1] CRAN (R 4.2.2)
#>  withr           2.5.0      2022-03-03 [1] CRAN (R 4.2.1)
#>  workflows     * 1.1.2      2022-11-16 [1] CRAN (R 4.2.2)
#>  workflowsets  * 1.0.0      2022-07-12 [1] CRAN (R 4.2.1)
#>  xfun            0.37       2023-01-31 [1] CRAN (R 4.2.2)
#>  yaml            2.3.7      2023-01-23 [1] CRAN (R 4.2.2)
#>  yardstick     * 1.1.0      2022-09-07 [1] CRAN (R 4.2.2)
#> 
#>  [1] C:/Users/bruno/AppData/Local/R/win-library/4.2
#>  [2] C:/Program Files/R/R-4.2.2/library
#> 
#> ──────────────────────────────────────────────────────────────────────────────
```

</details>
